### PR TITLE
fix(atomic): add hover effect on grid + handle grid click 

### PR DESCRIPTION
**Pull request by olamothe**
https://coveord.atlassian.net/browse/KIT-10

https://user-images.githubusercontent.com/1591893/137392216-42f53ece-1d66-46c4-8ab4-86ce572833ef.mov

### CODE REVIEW

File: packages/atomic/src/components/atomic-result-list/atomic-result-list.tsx [hunk 1]
```diff
@@ -181,6 +185,21 @@ export class AtomicResultList implements InitializableComponent {
           content={template.innerHTML}
         ></atomic-result>
       );
+
+      return this.display === 'grid' ? (
```
Comment (olamothe) [inline packages/atomic/src/components/atomic-result-list/atomic-result-list.tsx:189]: As soon as the results are displayed grid, any click on the "card" will trigger the default behaviour of opening the result.

**Review by ThibodeauJF — APPROVED** [review]
Looks great, although from the gif there seems to be a bit of a delay?

**Comment by olamothe** [comment]
> Looks great, although from the gif there seems to be a bit of a delay?

It's `0.3s ease-in` which mean it "start slow, finish fast"

File: packages/atomic/src/components/result-link/result-link.tsx [hunk 2]
```diff
@@ -12,18 +12,32 @@ export interface ResultLinkProps {
...
-    onTouchStart={() => interactiveResult.beginDelayedSelect()}
-    onTouchEnd={() => interactiveResult.cancelPendingSelect()}
-    target={target}
-  >
-    {children}
-  </a>
-);
+) => {
+  const stopPropagationAndProcess = (e: Event, process: () => void) => {
+    e.stopPropagation();
```
Comment (samisayegh) [inline packages/atomic/src/components/result-link/result-link.tsx:17]: Curious, what is `stopPropagation` preventing in this case?

Comment (olamothe) [reply packages/atomic/src/components/result-link/result-link.tsx:17]: If you have a link element within another link element, you don't want the click event to propagate in the DOM.

The click should only be handled once.

**Pull request merged**